# 01 — Data Loading
## Paddy Yield Predictor

In this notebook we just load the dataset and take a first look at it.
Nothing fancy — just checking what columns we have, the shape, missing values, and duplicates.

In [3]:
import pandas as pd
import sys
from pathlib import Path

# Add project root to path so we can import our own modules
PROJECT_ROOT = Path.cwd().parent if Path.cwd().name == 'notebooks' else Path.cwd()
sys.path.insert(0, str(PROJECT_ROOT))

from src.logger import get_logger
log = get_logger('01_data_loading')

log.info('Starting Data Loading notebook')

2026-08-18 15:02:32 | INFO     | 01_data_loading | Starting Data Loading notebook


In [4]:
DATA_PATH = PROJECT_ROOT / 'paddydataset.csv'

try:
    df = pd.read_csv(DATA_PATH)
    df.columns = df.columns.astype(str).str.strip()
    log.info(f'Dataset loaded — shape: {df.shape}')
except FileNotFoundError:
    log.error(f'CSV file not found at {DATA_PATH}. Please check the path.')
    raise
except Exception as e:
    log.error(f'Something went wrong while loading: {e}')
    raise

2026-08-18 15:02:33 | INFO     | 01_data_loading | Dataset loaded — shape: (2789, 45)


### Basic info about the dataset

In [5]:
print('Shape:', df.shape)
print('\nColumn names:')
print(df.columns.tolist())

Shape: (2789, 45)

Column names:
['Hectares', 'Agriblock', 'Variety', 'Soil Types', 'Seedrate(in Kg)', 'LP_Mainfield(in Tonnes)', 'Nursery', 'Nursery area (Cents)', 'LP_nurseryarea(in Tonnes)', 'DAP_20days', 'Weed28D_thiobencarb', 'Urea_40Days', 'Potassh_50Days', 'Micronutrients_70Days', 'Pest_60Day(in ml)', '30DRain( in mm)', '30DAI(in mm)', '30_50DRain( in mm)', '30_50DAI(in mm)', '51_70DRain(in mm)', '51_70AI(in mm)', '71_105DRain(in mm)', '71_105DAI(in mm)', 'Min temp_D1_D30', 'Max temp_D1_D30', 'Min temp_D31_D60', 'Max temp_D31_D60', 'Min temp_D61_D90', 'Max temp_D61_D90', 'Min temp_D91_D120', 'Max temp_D91_D120', 'Inst Wind Speed_D1_D30(in Knots)', 'Inst Wind Speed_D31_D60(in Knots)', 'Inst Wind Speed_D61_D90(in Knots)', 'Inst Wind Speed_D91_D120(in Knots)', 'Wind Direction_D1_D30', 'Wind Direction_D31_D60', 'Wind Direction_D61_D90', 'Wind Direction_D91_D120', 'Relative Humidity_D1_D30', 'Relative Humidity_D31_D60', 'Relative Humidity_D61_D90', 'Relative Humidity_D91_D120', 'Tras

In [6]:
# First 5 rows
df.head()

,Hectares,Agriblock,Variety,Soil Types,Seedrate(in Kg),LP_Mainfield(in Tonnes),Nursery,Nursery area (Cents),LP_nurseryarea(in Tonnes),DAP_20days,...,Wind Direction_D1_D30,Wind Direction_D31_D60,Wind Direction_D61_D90,Wind Direction_D91_D120,Relative Humidity_D1_D30,Relative Humidity_D31_D60,Relative Humidity_D61_D90,Relative Humidity_D91_D120,Trash(in bundles),Paddy yield(in Kg)
0,6,Cuddalore,CO_43,alluvial,150,75.0,dry,120,6,240,...,SW,W,NNW,WSW,72.0,78,88,85,540,35028
1,6,Kurinjipadi,ponmani,clay,150,75.0,wet,120,6,240,...,NW,S,SE,SSE,64.6,85,84,87,600,35412
2,6,Panruti,delux ponni,alluvial,150,75.0,dry,120,6,240,...,ENE,NE,NNE,W,85.0,96,84,79,600,36300
3,6,Kallakurichi,CO_43,clay,150,75.0,wet,120,6,240,...,W,WNW,SE,S,88.5,95,81,84,540,35016
4,6,Sankarapuram,ponmani,alluvial,150,75.0,dry,120,6,240,...,SSE,W,SW,NW,72.7,91,83,81,600,34044


In [7]:
# Data types and null counts
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2789 entries, 0 to 2788
Data columns (total 45 columns):
 #   Column                              Non-Null Count  Dtype  
---  ------                              --------------  -----  
 0   Hectares                            2789 non-null   int64  
 1   Agriblock                           2789 non-null   object 
 2   Variety                             2789 non-null   object 
 3   Soil Types                          2789 non-null   object 
 4   Seedrate(in Kg)                     2789 non-null   int64  
 5   LP_Mainfield(in Tonnes)             2789 non-null   float64
 6   Nursery                             2789 non-null   object 
 7   Nursery area (Cents)                2789 non-null   int64  
 8   LP_nurseryarea(in Tonnes)           2789 non-null   int64  
 9   DAP_20days                          2789 non-null   int64  
 10  Weed28D_thiobencarb                 2789 non-null   int64  
 11  Urea_40Days                         2789 no

In [8]:
# Quick check: missing values and duplicates
missing = df.isna().sum().sum()
dupes = df.duplicated().sum()

print(f'Total missing values : {missing}')
print(f'Duplicate rows       : {dupes}')

log.info(f'Missing values: {missing} | Duplicates: {dupes}')

2026-08-18 15:02:33 | INFO     | 01_data_loading | Missing values: 0 | Duplicates: 451


Total missing values : 0
Duplicate rows       : 451


In [9]:
# Basic statistics for numeric columns
df.describe().T

,count,mean,std,min,25%,50%,75%,max
Hectares,2789.0,3.717461,1.437777,1.00,3.00,4.00,5.00,6.00
Seedrate(in Kg),2789.0,92.936536,35.944420,25.00,75.00,100.00,125.00,150.00
LP_Mainfield(in Tonnes),2789.0,46.468268,17.972210,12.50,37.50,50.00,62.50,75.00
Nursery area (Cents),2789.0,74.349229,28.755536,20.00,60.00,80.00,100.00,120.00
LP_nurseryarea(in Tonnes),2789.0,3.717461,1.437777,1.00,3.00,4.00,5.00,6.00
DAP_20days,2789.0,148.698458,57.511072,40.00,120.00,160.00,200.00,240.00
Weed28D_thiobencarb,2789.0,7.434923,2.875554,2.00,6.00,8.00,10.00,12.00
Urea_40Days,2789.0,100.854729,39.006885,27.13,81.39,108.52,135.65,162.78
Potassh_50Days,2789.0,38.587250,14.924123,10.38,31.14,41.52,51.90,62.28
Micronutrients_70Days,2789.0,55.761922,21.566652,15.00,45.00,60.00,75.00,90.00


### What we found
- The dataset has **2789 rows** and **45 columns** (44 features + 1 target)
- No missing values — good, no imputation required at the data level
- There are **451 duplicate rows** — we'll remove them in the preprocessing step
- The target column is `Paddy yield(in Kg)`